semeio/jobs/overburden_timeshift/ots.py

```diff
@@ -172,34 +190,44 @@ def _create_surface(self, z=None):
         if z is None:
             z = self._surface.z
 
-        xstart = np.min(x)
-        ystart = np.min(y)
+        xstart = float(np.min(x))  # Casting to float for xtgeo
+        ystart = float(np.min(y))  # Casting to float for xtgeo
 
         if nx < 2 or ny < 2:
             raise RuntimeError("Cannot create IRAP surface if nx or ny is <2")
 
         xinc = (np.max(x) - xstart) / (nx - 1)
         yinc = (np.max(y) - ystart) / (ny - 1)
 
-        surf = Surface(
-            nx=nx, ny=ny, xinc=xinc, yinc=yinc, xstart=xstart, ystart=ystart, angle=0
+        surf_geo = xtgeo.RegularSurface(
+            ncol=nx,
+            nrow=ny,
+            xori=xstart,
+            yori=ystart,
+            xinc=xinc,
+            yinc=yinc,
+            rotation=0,
         )
 
         irap_x = np.empty(nx * ny)
         irap_y = np.array(irap_x)
 
-        for i in range(len(surf)):
-            irap_x[i], irap_y[i] = surf.getXY(i)
+        counter = 0
+        for x_index in range(1, nx + 1):
```

Comment:
Could technically do the same here as above: `for point, (x_index, y_index) in enumerate(product(`